src/app/routes/utils/constructPageFetchUrl/index.ts

```diff
@@ -0,0 +1,94 @@
+import Url from 'url-parse';
+import pipe from 'ramda/src/pipe';
+import getEnvironment from '#app/routes/utils/getEnvironment';
+import { getUrlPath } from '../../../lib/utilities/urlParser';
+import handleError from '../handleError';
+import { Services, Variants } from '../../../models/types/global';
+
+interface UrlConstructParams {
+  pathname: string;
+  pageType: 'article' | 'cpsAsset' | 'topic' | 'live';
+  service: Services;
+  variant?: Variants;
+  page?: string;
+}
+
+const removeAmp = (path: string) => path.split('.')[0];
+const getArticleId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}o)/)?.[1];
+const getTipoId = (path: string) => path.match(/(c[a-zA-Z0-9]{10}t)/)?.[1];
+const getCpsId = (path: string) => path;
+
+const getId = (pageType: string) => {
+  let getIdFunction;
+  switch (pageType) {
+    case 'cpsAsset':
+      getIdFunction = getCpsId;
+      break;
+    case 'topic':
+    case 'live':
+      getIdFunction = getTipoId;
+      break;
+    default:
+      getIdFunction = getArticleId;
+      break;
+  }
+  return pipe(getUrlPath, removeAmp, getIdFunction);
+};
+
+const constructPageFetchUrl = ({
+  pathname,
+  pageType = 'topic',
```

Comment:
Should we default this? In the BFF we throw an exception if the pageType doesn't match

Reply:
I think I defaulted to this, because I thought the BFF would default to topic originally. If this has changed, shall we just throw an exception with a 404 is pageType is non-existent?

Reply:
Yeah with the current logic, if there is no pageType it would assume that the ID would be an article, and it would check to see if the ID is valid: https://github.com/bbc/simorgh/blob/043c16b6edf467482171b9fcce9a39058408c241/src/app/routes/utils/constructPageFetchUrl/index.ts#L42

If it's not it would 500 [here](https://github.com/bbc/simorgh/blob/043c16b6edf467482171b9fcce9a39058408c241/src/app/routes/utils/constructPageFetchUrl/index.ts#L62). But if by chance the ID is valid then yeah the BFF would throw the error.

Reply:
How about something like this:
```jsx
const getId = (pageType: PageTypes) => {
  let getIdFunction;
  switch (pageType) {
    case PAGE_TYPES.ARTICLE:
      getIdFunction = getArticleId;
      break;
    case PAGE_TYPES.CPS_ASSET:
      getIdFunction = getCpsId;
      break;
    case PAGE_TYPES.LIVE:
    case PAGE_TYPES.TOPIC:
      getIdFunction = getTipoId;
      break;
    default:
      getIdFunction = () => null;
      break;
  }
  return pipe(getUrlPath, removeAmp, getIdFunction);
};
```
Then we can get rid of the default topic, and simorgh would now throw a 500 as the id is falsy, and no call would be made to the BFF.

Reply:
Yes, makes a lot of sense! We might be able to add the ID fetching logic for homepage into this -  https://github.com/bbc/simorgh/blob/5c6d2aba9814d7909bb7e0cf2e9d013f528b2d64/src/app/routes/homePage/getInitialData/index.ts#L31